src/openms/source/METADATA/SpectrumLookup.cpp

```diff
@@ -311,11 +319,27 @@ namespace OpenMS
           return -1;
         }
       }
+      else if (matches.size() == 2)
+      {
+        try
+        {
+          int value = String(matches[0]).toInt() * 1000 + String(matches[1]).toInt();
```

Comment:
TODO: check that `String(matches[1]).toInt() < 1000` and warn/abort/fail if not.